nixos/modules/installer/tools/tools.nix

```diff
@@ -44,23 +35,141 @@ let
     inherit (config.system) configurationRevision;
     json = builtins.toJSON ({
       nixosVersion = config.system.nixos.version;
-    } // optionalAttrs (config.system.nixos.revision != null) {
+    } // lib.optionalAttrs (config.system.nixos.revision != null) {
       nixpkgsRevision = config.system.nixos.revision;
-    } // optionalAttrs (config.system.configurationRevision != null) {
+    } // lib.optionalAttrs (config.system.configurationRevision != null) {
       configurationRevision = config.system.configurationRevision;
     });
     manPage = ./manpages/nixos-version.8;
   };
 
-  inherit (pkgs) nixos-enter;
+  nixos-install = pkgs.nixos-install.override { nix = config.nix.package; };
+  nixos-rebuild = pkgs.nixos-rebuild.override { nix = config.nix.package; };
+
+  defaultConfigTemplate = ''
+    # Edit this configuration file to define what should be installed on
+    # your system. Help is available in the configuration.nix(5) man page, on
+    # https://search.nixos.org/options and in the NixOS manual (`nixos-help`).
+
+    { config, lib, pkgs, ... }:
+
+    {
+      imports =
+        [ # Include the results of the hardware scan.
+          ./hardware-configuration.nix
+        ];
+
+    $bootLoaderConfig
```

Comment:
We were debating the other day that this section is kind of hardware related. But this another pull request.

Reply:
I still kinda want to rewrite the whole thing tbh

Reply:
What do you want to rewrite this to? We are also working on a replacement as well.